SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -150,10 +170,23 @@ bool FLocalDeploymentManager::TryStartLocalDeployment(FString LaunchConfig, FStr
 
 	bStartingDeployment = true;
 
+	// Stop the currently running service if the runtime IP is to be exposed, but is different from the one specified
+	if (ExposedRuntimeIP != RuntimeIPToExpose)
+	{
+		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Settings for exposing runtime IP have changed since service startup. Restarting service to reflect changes."));
+		TryStopSpatialService();
+	}
+
 	// If the service is not running then start it.
 	if (!bSpatialServiceRunning)
 	{
-		TryStartSpatialService();
+		if (RuntimeIPToExpose.IsEmpty()) {
+			TryStartSpatialService();
```

Comment:
Won't this internally check for a null string? So the outcome of calling it with RuntimeIPToExpose and without is the same?

Reply:
Yes, running `TryStartSpatialService(RuntimeIPToExpose)` would also work - technically the `IsEmpty()` check is not necessary here. That would, however, only work as long as the default value for the optional `RuntimeIPToExpose` argument (which is the empty string) is the same for both the `TryStartLocalDeployment` and `TryStartSpatialService` methods. I figured it would be better to program defensively at the cost of code compactness. What are your thoughts on this?

Reply:
I'm generally in favour of fewer branches and either doing these checks outside or inside a function (usually inside, unless performance is a concern), but not both. Not hugely fussed though.